sql/hive/src/main/scala/org/apache/spark/sql/hive/client/ClientWrapper.scala

```diff
@@ -167,6 +169,8 @@ private[hive] class ClientWrapper(
       } else {
         logInfo("Attempting to login to Kerberos" +
           s" using principal: ${principalName} and keytab: ${keytabFileName}")
+        // make sure the core-site.xml is provided in your classpath
```

Comment:
Better than this would be to update `docs/sql-programming-guide.md`. It currently only mentions `hive-site.xml`, when really you need `core-site.xml` and `hdfs-site.xml` too (otherwise kerberos, in your case, or things like HDFS HA won't work).

The other changes in this file can be reverted, too.